nsy/nsy_hip/nsy_hip.cu

```diff
@@ -0,0 +1,79 @@
+#ifdef _MSC_VER
+#include "msvc_defines.h"
+#endif
+#include <hip/hip_runtime.h>
+#include <iostream>
+#include <locale>
+#include <string>
+#include <vector>
+
+#define HIP_CHECK(status)                                                      \
+  if (status != hipSuccess) {                                                  \
+    fprintf(stderr, "error: '%s' at %s:%d\n", hipGetErrorString(status),       \
+            __FILE__, __LINE__);                                               \
+    exit(0);                                                                   \
+  }
+
+__global__ void kernel(const wchar_t *in, wchar_t *out, const size_t dim,
+                       const wchar_t mark) {
+  const int idx = hipThreadIdx_x * dim + hipThreadIdx_y;
+  if (hipBlockIdx_x) {
+    out[idx * 2 + 1] = mark;
+  } else {
+    out[idx * 2] = in[idx];
+  }
+}
+
+wchar_t get_mark(char *s) {
+  const char a = s[0];
+  const char b = s[1];
+  const bool u = b != '\0' && b == 'u';
+  if (a == 'q') {
+    return u ? L'\x00BF' : L'\xFF1F';
```

Comment:
![image](https://github.com/sdbx/Utils/assets/83723320/e7f308d9-9724-4f7e-9949-cd4c46af725e)
Very wise to use the ternary operator rather than two if statements!